### ISSUE TITLE
Check incumbents in election endpoint

### ISSUE THREAD

**Issue by PaulClark2** [fecgov/fec-cms#2015]
Make sure incumbents are marked correctly in /elections/ API endpoint.

Previous research was done on /elections/search/ but we don't use the incumbent data in that endpoint on the front end, and it uses different logic. I put in an issue to remove incumbent data from /elections/search/ for the time being: https://github.com/fecgov/openFEC/issues/3247

**Comment by llienfec** [comment]
Current blocker: learning how to export the API's JSON output into google sheets (i.e. human readable format).  Did a little research and found a google sheets script that is almost working.  Will work on it a little more but may need help.

**Comment by llienfec** [comment]
No longer blocked - Dorothy sent a website that converts the JSON API results to spreadsheets.  It definitely helps, but is still cumbersome with paginated JSON results.  It would be good to find a better solution for people to use the API results in a spreadsheet format.

Made really good headway on this issue and will have the research finished tomorrow.

**Comment by llienfec** [comment]
## Method
To check the incumbent flags in the /elections/ endpoint, I pulled data from the /elections/search/ API endpoint (had all the needed data).  Compared incumbents listed by district to external data sources, looking for discrepancies and researching the differences.  It was a lot of data, so apologies for any oversights/spelling issues! 

## Data
My full research notes are here: 
:lock: https://docs.google.com/spreadsheets/d/1N-v4NSycB6JE54BIRmrvWPI74F-GA8C82Nb5lgRuEcM/edit?usp=sharing :lock: 
@rhoughfec is good at interpreting my massive spreadsheets in case anything needs deciphering while I'm out of the office next week.

## Findings summary
- Most common error: districts that had special elections, resignations, deaths are out of date.
- "Null" for the incumbent value is used inconsistently.  Seems like it should be used when the seat is vacant (examples: MI 13, TX 27) or maybe when the person in the office was appointed and not elected.
- Pennsylvania's redistricting moved incumbents around.  The 1:1 ratio between districts and incumbents doesn't work 2 PA districts.
- Less common error: winner from 2016 wasn't flagged properly

### House changes (not including PA)

State  |  District |  API incumbent  | Incumbent should be | Reason
------ | --------- | ---------------- | ----------------------- | ---------
AZ | 08 | Null | Debbie Lesko | Won special election
GA | 06 | Thomas Price | Karen Handel | TP appointed to HHS; KH won special
KS | 04 | Null | Ron Estes | Pompeo appointed to State; RE won special
MT | 08 | Null | Greg Gianforte| Zinke appointed to Interior; GG won special
NC | 02 | Null | George Holding | 
NC | 13 | George Holding | Ted Bud | 
NY | 25 | Louise Slaughter | Null | Incumbent deceased; seat vacant
OH | 12 | Patrick Tiberi | Null | Resigned Oct 2017; will have an incumbent to flag after 8/7/18 special
OK | 01 | James Bridenstine | Null | Appointed to NASA; seat vacant
UT | 03 | Jason Chaffetz | John Curtis| Resigned April 2017; JC won special

### House updates for PA redistricting
Note districts 03 and 17 have 2 incumbents running. 

New district |  Incumbent should be | Old district
-------- | ---------------- | ----------------------- 
01 | Brian Fitzpatrick | 8
02 | Brendan Boyle | 13
**03** | Bob Brady | 1
**03** | Dwight Evans | 2
04 | Open seat | 
05 | Vacant | 
06 | Ryan Costello | 6
07 | Vacant | 
08 | Matt Cartwright | 17
09 | Lou Barletta | 8
10 | Scott Perry | 4
11 | Lloyd Smucker | 16
12 | Tom Marino | 10
13 | Bill Schuster | 9
14 | Open | 
15 | Glenn Thompson | 5
16 | Mike Kelly | 3
**17** | Keith Rothfus | 12
**17** | Conor Lamb | 18
18 | Mike Doyle | 14

### Senate updates
Senate class | State | API incumbent | Incumbent should be | Reason
------ | --------- | ---------------- | ----------------------- | ---------
Class 1 - 2018 | MN | Tina Flint Smith | Amy Klobuchar | AK holds the Class 1 seat.  TFS holds the Class 2 seat
Class 2 - 2020 | AL | Jeff Sessions | Doug Jones | JS appointed to DOJ; DJ won special
Class 2 - 2020 | AL | Al Franken | Tina Flint Smith | Won special in 2018 for Class 2 seat
Class 2 - 2020 | MS | Thad Cochran | Null | Resigned and Cindy Hyde-Smith was appointed by governor
Class 3 - 2022 | IL | Null | Tammy Duckworth | 
Class 3 - 2022 | IN | Null | Todd Young | 
Class 3 - 2022 | WI | Null | Ron Johnson |

**Comment by llienfec** [comment]
@PaulClark2 This research is finished, so I'm closing the issue.  We may need an implementation issue to fix some of the incumbents in the charts above.  Thanks!

**Comment by lbeaufort** [comment]
Reopening this issue - we need to test /elections/ to see the data pulled onto the election profile pages. Also can test the data from the front-end if it's easier. (example: https://www.fec.gov/data/elections/house/AZ/08/2018/)

**Comment by llienfec** [comment]
## /elections/ endpoint research findings - House candidates

**Finding 1:** Almost all incumbent markers are correct.  
**Finding 2:** Incorrect incumbents are from recent specials (2017 & 2018)

State | District | Incumbent should be 
-- | -- |  -- 
GA | 06 | Karen Handel
PA | 18 | Conor Lamb
TX | 27 | Micheal Cloud
UT | 03 | John Curtis

**Comment by llienfec** [comment]
## /elections/ endpoint research findings - Senate candidates

- All incumbent markers appear to be correct for states that have elections in 2018.
- When searching for future senate elections in 2020 and 2022, the search does not return results on the site and in the endpoint.

Findings for both House and Senate were added to the google doc.
:lock: https://docs.google.com/spreadsheets/d/1N-v4NSycB6JE54BIRmrvWPI74F-GA8C82Nb5lgRuEcM/edit?usp=sharing :lock:

**Comment by lbeaufort** [comment]
@llienfec thanks for taking a look! I entered an issue for missing 2020 data in /elections/: https://github.com/fecgov/openFEC/issues/3243

**Comment by llienfec** [comment]
Closing this research issue.  We successfully tested /elections/search/ and hid the incumbent information in favor of the info from the /elections/ endpoint.

- We tested /elections/ endpoint for incumbent markers for House committees.  Findings are above.  
- We need to add 2020/2022 data to the database (https://github.com/fecgov/openFEC/issues/3243) so that we can test the S & P data (https://github.com/fecgov/openFEC/issues/3261)
- Added a future work ticket to fix incumbent flags  for once the work above is done (https://github.com/fecgov/openFEC/issues/3262)
